10up/wp-scaffold
# Removing lerna in favor of npm workspaces (npm >=7)

### Description of the Change



Lerna was added to the scaffold to allow us to better handle npm dependencies in the new scaffold structure but, we're only using `lerna` to handle installing dependencies. However, npm workspaces have been introduced in npm 7 and we no longer need lerna in the scaffold.

Lerna also requires following a different workflow for adding deps (`lerna add`) which can be a bit cumbersome to unfamiliar engineers.

By moving to `npm workspaces` we reduce one dependency (which isn't very actively maintained at this point) and we simplify the overall workflow.

Instead of `lerna.json`, we juts need to add  `workspaces` in package.json.

Adding/remove deps is much more familiar to everyone as you only need to use a flag to specify the package:
`npm install --save prop-types -w=package-name`

Closes #

### Alternate Designs



### Possible Drawbacks


npm workspaces can't run scripts in parallel so blocking scripts (like watch/start) will prevent npm running the rest of the commands. To workaround this, `npm-run-all` has been introduced.

### Verification Process



- Remove `node_modules`
- Make sure you're using npm >= 7
- npm install
- npm run build
- npm run watch
- npm run start


### Checklist:



- [x My code follows the code style of this project.



### Changelog Entry



Changed - Removing `lerna` in favor of `npm workspaces`

### Credits


Props @nicholasio
